c7n/resources/rds.py

```diff
@@ -1648,12 +1648,9 @@ def validate(self):
 
     def process(self, resources):
         c = local_session(self.manager.session_factory).client('rds')
-
         for r in resources:
-            param = {}
-            for update in self.data.get('update'):
-                if r[update['property']] != update['value']:
-                    param[update['property']] = update['value']
+            param = {
+                update['property']: update['value'] for update in self.data.get('update')}
```

Comment:
this change looks like it was intended to be stylistic, but lost a feature of not causing an api call if there was no change to the resource.

Reply:
It did become a stylistic change but the `r[update['property']] != update['value']` bit was actually what was causing this bug because the api call names did not match the resource param names. 
e.g.
PerformanceInsightsEnabled on resource and EnablePerformanceInsights for the enable call

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/rds.html#RDS.Client.describe_db_instances
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/rds.html#RDS.Client.modify_db_instance

Reply:
My thoughts around removing the check were for parity with the modify-db-cluster action. Though it may be useful to provide a mapping of property to api call for translation instead of removing this check.

Reply:
yeah.. we should do the mapping for the property names that are a delta, to avoid the api call ideally, in this context it might be overkill we could just check more cleanly .get() or in that the property in update exists and is a delta. then the keyerror here would go away, we'd get spurious updates for performance insights in particular though its a tradeoff on mapping maintenance and optimal behavior that I'm fine with.